march_hardware/include/march_hardware/Encoder.h

```diff
@@ -58,23 +51,48 @@ class Encoder
   /** @brief Override comparison operator */
   friend bool operator==(const Encoder& lhs, const Encoder& rhs)
   {
-    return lhs.slaveIndex == rhs.slaveIndex && lhs.totalPositions == rhs.totalPositions &&
-           lhs.upperSoftLimitIU == rhs.upperSoftLimitIU && lhs.lowerSoftLimitIU == rhs.lowerSoftLimitIU &&
-           lhs.upperHardLimitIU == rhs.upperHardLimitIU && lhs.lowerHardLimitIU == rhs.lowerHardLimitIU &&
-           lhs.zeroPositionIU == rhs.zeroPositionIU && lhs.safetyMarginRad == rhs.safetyMarginRad;
+    return lhs.slave_index_ == rhs.slave_index_ && lhs.total_position_ == rhs.total_position_ &&
+           lhs.upper_soft_limit_iu_ == rhs.upper_soft_limit_iu_ &&
+           lhs.lower_soft_limit_iu_ == rhs.lower_soft_limit_iu_ && lhs.upper_limit_iu_ == rhs.upper_limit_iu_ &&
+           lhs.lower_limit_iu_ == rhs.lower_limit_iu_ && lhs.zero_position_iu_ == rhs.zero_position_iu_;
   }
   /** @brief Override stream operator for clean printing */
-  friend ::std::ostream& operator<<(std::ostream& os, const Encoder& encoder)
+  friend std::ostream& operator<<(std::ostream& os, const Encoder& encoder)
   {
-    return os << "slaveIndex: " << encoder.slaveIndex << ", "
-              << "totalPositions: " << encoder.totalPositions << ", "
-              << "upperHardLimit: " << encoder.upperHardLimitIU << ", "
-              << "lowerHardLimit: " << encoder.lowerHardLimitIU << ", "
-              << "upperSoftLimit: " << encoder.upperSoftLimitIU << ", "
-              << "lowerSoftLimit: " << encoder.lowerSoftLimitIU << ", "
-              << "zeroPositionIU: " << encoder.zeroPositionIU << ", "
-              << "safetyMarginRad: " << encoder.safetyMarginRad;
+    return os << "slaveIndex: " << encoder.slave_index_ << ", "
+              << "totalPositions: " << encoder.total_position_ << ", "
+              << "upperHardLimit: " << encoder.upper_limit_iu_ << ", "
+              << "lowerHardLimit: " << encoder.lower_limit_iu_ << ", "
+              << "upperSoftLimit: " << encoder.upper_soft_limit_iu_ << ", "
+              << "lowerSoftLimit: " << encoder.lower_soft_limit_iu_ << ", "
+              << "zeroPositionIU: " << encoder.zero_position_iu_;
   }
+
+  static const size_t MIN_RESOLUTION = 1;
+  static const size_t MAX_RESOLUTION = 32;
+
+  static constexpr double MAX_IU_PER_RAD_DIFFERENCE = 0.1;
```

Comment:
Why a maximum? We already know what the IU per rad is for all the different joints. IU per rad should be constant and if it is not and something is going wrong somewhere.

Reply:
This maximum is used to give a warning when the IU per rad calculated from the encoder positions is 10% different than the IU per rad calculated from the given limits. So indeed something is wrong when the two values are different (which is happening right now, but that is another issue)